protocols/gossipsub/src/protocol.rs

```diff
@@ -319,6 +400,76 @@ impl Decoder for GossipsubCodec {
     }
 }
 
+/// Signs a gossipsub message, adding the key and signature fields.
+fn sign_message(keypair: &Keypair, message: &mut rpc_proto::Message) -> Result<(), io::Error> {
```

Comment:
TODO mxinden: Why a function and not a method?

Reply:
To save you some time. Originally these were inlined, I pulled them out to their own logic functions and they were easier to test. 

There's no real reason why they are not methods, except that the verify message function requires no struct fields of the codec. 

None the less, I've re-written these as methods, because, why not :)